kotsadm/pkg/handlers/gitops.go

```diff
@@ -105,6 +110,129 @@ func DisableAppGitOps(w http.ResponseWriter, r *http.Request) {
 	JSON(w, http.StatusNoContent, "")
 }
 
+func InitGitOpsConnection(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Headers", "content-type, origin, accept, authorization")
+
+	if r.Method == "OPTIONS" {
+		w.WriteHeader(http.StatusOK)
+		return
+	}
+
+	if err := requireValidSession(w, r); err != nil {
+		logger.Error(err)
+		return
+	}
+
+	appID := mux.Vars(r)["appId"]
+	clusterID := mux.Vars(r)["clusterId"]
+
+	a, err := app.Get(appID)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	d, err := downstream.Get(clusterID)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	downstreamGitOps, err := gitops.GetDownstreamGitOps(a.ID, d.ClusterID)
+	if err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if downstreamGitOps == nil {
+		logger.Error(errors.New("downstream gitops not found"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if downstreamGitOps.Format != "single" {
+		logger.Error(errors.New("unsupported gitops format"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if err := gitops.TestGitOpsConnection(downstreamGitOps); err != nil {
+		logger.Error(err)
+		err = gitops.SetGitOpsError(a.ID, d.ClusterID, err.Error())
+		if err != nil {
+			logger.Error(err)
+		}
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if err := gitops.SetGitOpsError(a.ID, d.ClusterID, ""); err != nil {
+		logger.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	go func(appID, clusterID, appSlug, appName, downstreamName string) {
+		currentVersion, err := downstream.GetCurrentVersion(appID, clusterID)
```

Comment:
This should work like checking for updates works or how setting registry credentials works.  even if we can't show a status update, a second request to this handler should fail if the first one is still running